cmd/main.go

```diff
@@ -55,32 +56,46 @@ func main() {
 				Action: func(cCtx *cli.Context) error {
 					var id principal.Signer
 					var err error
-					if cCtx.String("private-key") == "" {
-						id, err = ed25519.Generate()
-						if err != nil {
-							return fmt.Errorf("generating ed25519 key: %w", err)
-						}
-						log.Errorf("Server ID is not configured, generated one for you: %s", id.DID().String())
-					} else {
-						id, err = ed25519.Parse(cCtx.String("private-key"))
-						if err != nil {
-							return fmt.Errorf("parsing private key: %w", err)
+
+					pkstr := cCtx.String("private-key")
+					if pkstr == "" {
+						if os.Getenv("PRIVATE_KEY") != "" {
```

Comment:
suggestion: instead of doing this you can just set the EnvVars property on the CLI flag
https://cli.urfave.org/v2/examples/flags/#values-from-the-environment

Reply:
Oh I had no idea you could do that! Awesome.